lib/mp4/probe.js

```diff
@@ -160,16 +164,18 @@ compositionStartTime = function(timescales, fragment) {
     // The spec states that track run samples contained within a `traf` box are contiguous, but
     // it does not explicitly state whether the `traf` boxes themselves are contiguous.
     // We will assume that they are, so we only need the first to calculate start time.
-    var parsedTraf = mp4Inspector.parseTraf(trafBoxes[0]);
-
-    for (var i = 0; i < parsedTraf.boxes.length; i++) {
-      if (parsedTraf.boxes[i].type === 'tfhd') {
-        trackId = parsedTraf.boxes[i].trackId;
-      } else if (parsedTraf.boxes[i].type === 'tfdt') {
-        baseMediaDecodeTime = parsedTraf.boxes[i].baseMediaDecodeTime;
-      } else if (parsedTraf.boxes[i].type === 'trun' && parsedTraf.boxes[i].samples.length) {
-        compositionTimeOffset = parsedTraf.boxes[i].samples[0].compositionTimeOffset || 0;
-      }
+    var tfhd = parseTfhd(findBox(trafBoxes[0], ['tfhd'])[0]);
+    var trun = parseTrun(findBox(trafBoxes[0], ['trun'])[0]);
+    var tfdt = parseTfdt(findBox(trafBoxes[0], ['tfdt'])[0]);
```

Comment:
Not sure if this is a case we will/have run into, but it looks like if those boxes are not in the segment, we will throw exceptions here (findBox will return an empty array, first element will be `undefined` and each parse function would try to reference the first element of `undefined`). In the past it seemed we just didn't have the result.